src/main/java/lotto/domain/LottoNumbers.java

```diff
@@ -3,37 +3,70 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 public class LottoNumbers {
-	private static final int MATCHED_NUMBER_PLUS = 1;
-	private static final int NON_MATCHED_NUMBER = 0;
+	private static final String ERROR_DUPLICATE_LOTTO_NUMBER = "[ERROR] 로또 번호는 중복 될 수 없습니다.";
+	private static final String ERROR_LOTTO_NUMBER_LENGTH = "[ERROR] 로또 번호는 6자리 숫자를 가져야 합니다.";
 	private static final String DELIMITER = ", ";
-	private final List<Integer> lottoNumbers;
+	public static final int MATCHED_NUMBER = 1;
+	private static final int NON_MATCHED_NUMBER = 0;
+	public static final int MIN_RANGE_VALUE = 0;
+	public static final int MAX_RANGE_VALUE = 6;
+	private final List<LottoNumber> lottoNumbers;
 
-	public LottoNumbers(int... lottoNumbers) {
+	public LottoNumbers(final int... lottoNumbers) {
 		this(Arrays.stream(lottoNumbers)
-			.boxed()
+			.mapToObj(LottoNumber::new)
 			.collect(Collectors.toList()));
 	}
 
-	public LottoNumbers(List<Integer> lottoNumbers) {
-		this.lottoNumbers = Collections.unmodifiableList(new ArrayList<>(lottoNumbers));
+	public LottoNumbers(final String lottoNumbers) {
+		this(validateStringLottoNumbers(lottoNumbers));
+	}
+
+	public LottoNumbers(final List<LottoNumber> lottoNumbers) {
+		validateLottoNumberLength(lottoNumbers);
+		this.lottoNumbers = new ArrayList<>(lottoNumbers);
 	}
 
-	public int isMatch(int winningNumber) {
+	public int ifMatchCount(LottoNumber winningNumber) {
 		if (lottoNumbers.contains(winningNumber)) {
-			return MATCHED_NUMBER_PLUS;
+			return MATCHED_NUMBER;
 		}
 		return NON_MATCHED_NUMBER;
 	}
 
-	public String StringValues() {
-		String values = "";
-		for (Integer lottoNumber : lottoNumbers) {
-			values += String.valueOf(lottoNumber) + DELIMITER;
+	public List<LottoNumber>  getLottoNumbers() {
+		return Collections.unmodifiableList(lottoNumbers);
+	}
+
+	private static List<LottoNumber> validateStringLottoNumbers(String lottoNumbers) {
+		String[] stringNumberArray = lottoNumbers.split(DELIMITER);
+		validateDuplicate(stringNumberArray);
+		return Arrays.stream(stringNumberArray)
+			.map(LottoNumber::new)
+			.collect(Collectors.toList());
+	}
+
+	private static void validateDuplicate(String[] stringNumberArray) {
+		Set<String> duplicateCheck = new HashSet<>();
+		for (String number : stringNumberArray) {
+			duplicateCheck.add(number);
+		}
+		if (duplicateCheck.size() < MAX_RANGE_VALUE) {
+			throw new IllegalArgumentException(ERROR_DUPLICATE_LOTTO_NUMBER);
 		}
-		return values.substring(0, values.length()-2);
 	}
+
+	private void validateLottoNumberLength(List<LottoNumber> lottoNumbers) {
+		if (lottoNumbers.size() < MAX_RANGE_VALUE) {
```

Comment:
로또 번호가 6개 미만이면 검증에 실패하지 않을까요?
아래와 같이 수정해보는건 어떨까요?
```suggestion
		if (lottoNumbers.size() == LOTTO_NUMBER_COUNT) {
```

Reply:
안녕하세요 새벽에 자꾸 질문을 드리게 되네요 죄송합니다. ㅠ
그 해당 리뷰해주신부분은 로또번호가 6개 미만으로 검출되면 에러가 발생하도록 작성한 코드입니다.
그래서 6개 미만일경우에 에러가 발생하는게 맞는데 수정해주신 코드로 바꾸게 되면 6개 정상적으로 생성되었을때 에러가 발생하는 것 같습니다.
혹시 제가 리뷰해주신 부분에서 잘 못 이해하고 있는부분이 있을까요???

Reply:
제가 반대로 이야기드렸네요..  😅 
6개 이상일때를 이야기드린 것이었습니다 :)
로또 번호는 항상 6개로 고정되어있어야 하지 않을까요? 🤔

Reply:
아하 넵 수정했습니다!